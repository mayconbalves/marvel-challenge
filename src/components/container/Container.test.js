import { render, screen, fireEvent } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import Container from './'

describe('<Container />', () => {
  const mockStore = configureStore([thunk])
  let store
  store = mockStore({
    charactersReducer: { characters: [] }
  })

  const children = document.body.innerHTML = `
  <div>
    <h1>children</h1>
  </div>
`

  let props = {
    children,
    color: '#fff'
  }

  it('should render correct title element', () => {
    const { container } = render(
      <MemoryRouter>
        <Provider store={store}>
          <Container {...props} />
        </Provider>
      </MemoryRouter>
    )

    expect(
      screen.getByRole('heading', { Name: 'children' })
    ).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })
})
