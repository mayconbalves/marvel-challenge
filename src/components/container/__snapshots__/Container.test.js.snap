// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> should render correct title element 1`] = `
<div
  class="sc-bdfBwQ kSwYoH"
  color="#fff"
>
  
  &lt;div&gt;
    &lt;h1&gt;children&lt;/h1&gt;
  &lt;/div&gt;

</div>
`;
